ariadne/graphql.py

```diff
@@ -136,6 +145,12 @@ def graphql_sync(
 
             if callable(validation_rules):
                 validation_rules = validation_rules(context_value, document, data)
+            if not introspection:
+                validation_rules = (
+                    list(validation_rules) + [IntrospectionDisabledRule]
+                    if validation_rules
+                    else [IntrospectionDisabledRule]
+                )
```

Comment:
Looks too boiler-platey being repeated three times in file, and having `[IntrospectionDisabledRule]` six times total.

```suggestion
            if not introspection:
                validation_rules = disable_query_introspection(validation_rules)


def disable_query_introspection(validation_rules: Optional[ValidationRules]) -> ValidationRules:
    if validation_rules:
        return validation_rules = [IntrospectionDisabledRule]
    return [IntrospectionDisabledRule]
```